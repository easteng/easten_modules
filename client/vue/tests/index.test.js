import pkg from '../package.json';

const testDist = process.env.LIB_DIR === 'dist';

describe('snweb-module dist files', () => {
  // https://github.com/ant-design/ant-design/issues/1638
  // https://github.com/ant-design/ant-design/issues/1968
  it('exports modules correctly', () => {
    const snwebModule = testDist ? require('../dist/snweb-module') : require('../components'); // eslint-disable-line global-require
    expect(Object.keys(snwebModule)).toMatchSnapshot();
  });

  // https://github.com/ant-design/ant-design/issues/1970
  // https://github.com/ant-design/ant-design/issues/1804
  if (testDist) {
    it('should have snweb-module.version', () => {
      const snwebModule = require('../dist/snweb-module'); // eslint-disable-line global-require
      expect(snwebModule.version).toBe(pkg.version);
    });
  }
});
