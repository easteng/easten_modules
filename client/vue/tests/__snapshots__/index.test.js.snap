// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`snweb-module dist files exports modules correctly 1`] = `
Array [
  "Base",
  "version",
  "Button",
  "SmSystemUsers",
  "SmSystemRoles",
  "SmSystemOrganizationUnits",
  "SmSystemOrganizationUnitTreeSelect",
  "SmSystemDataDictionaryTreeSelect",
  "SmSystemDataDictionaries",
  "default",
]
`;
